internal/venice-test-common/src/main/java/com/linkedin/venice/integration/utils/VeniceRouterWrapper.java

```diff
@@ -164,8 +186,19 @@ protected void internalStop() throws Exception {
 
   @Override
   protected void newProcess() {
-    List<ServiceDiscoveryAnnouncer> d2Servers =
-        D2TestUtils.getD2Servers(zkAddress, "http://localhost:" + getPort(), "https://localhost:" + getSslPort());
+    List<ServiceDiscoveryAnnouncer> d2Servers = D2TestUtils.getD2ServersForRouter(
+        zkAddress,
+        d2ClusterName,
+        "http://localhost:" + getPort(),
```

Comment:
Nitpick: I think we should call `getHost()` here rather than hardcode "localhost". Right now this is always equivalent, but there are some discussions around making the test infra capable of returning process wrappers that point to external resources... that is the intent of this wrapper design, in any case.

Reply:
There are two instances where we are using localhost - one is this one and another is before the constructor. So, we will need some slightly bigger changes to use `Utils.getHostName()` instead. In this instance, it is possible to use `getHost()` since we are operating on an instance of `VeniceRouterWrapper`

I've updated this case - but left the other one still using `localhost`